contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -190,15 +234,31 @@ abstract contract AccessControlDefaultAdminRules is IAccessControlDefaultAdminRu
     }
 
     /**
-     * @dev See {acceptDefaultAdminTransfer}.
+     * @dev See {beginDefaultAdminDelayChange}.
      *
      * Internal function without access restriction.
      */
-    function _acceptDefaultAdminTransfer() internal virtual {
-        require(_hasDefaultAdminTransferDelayPassed(), "AccessControl: transfer delay not passed");
-        _revokeRole(DEFAULT_ADMIN_ROLE, defaultAdmin());
-        _grantRole(DEFAULT_ADMIN_ROLE, pendingDefaultAdmin());
-        _resetDefaultAdminTransfer();
+    function _beginDefaultAdminDelayChange(uint48 newDefaultAdminDelay) internal virtual {
+        require(
+            !_hasScheduledPassed(defaultAdminDelayChangeSchedule()),
+            "AccessControl: can't change past scheduled change"
+        );
```

Comment:
`defaultAdminDelayChangeSchedule` is only cleared on `_reset`, `_reset` is only called on `cancel`. It seems like it is only possible to change the delay once.

Reply:
After refactoring with the suggested style changes, this is still sort of a problem. However, my recent updates include the following:

- I renamed this error to `AccessControl: can't change virtual delay` instead. Signaling that the value is virtual and not yet materialized
- Although a `_reset*` is required to make the current delay materialized, the reset is now also called from `cancel*`, removing the possibility for an actual cancellation.
- Under the same logic, this `require(..., ...)` can be replaced by a `_materializeDefaultAdminDelayChange` by splitting the current `_resetDefaultAdminDelayChange` in two.

We either keep the `require` or add `_materialize*()`. I'm ok for the latter, but will think more about it.